### ISSUE TITLE
In Browse mode, the screen reader announces 'link' on every control and text under 'Browse App' section. 

### ISSUE THREAD

**Issue by msftedad** [nvaccess/nvda#15679]
### Steps to reproduce:

1. Open the [URL](https://admin.microsoft.com/adminportal/home?#/Settings/IntegratedApps) on edge dev browser. 
2. The 'Integrated apps' page will be opened. 
3. Turn on NVDA using 'Ctrl+ Alt+N' keys. 
4. Navigate to the 'Get Apps' button using tab key and activate it. 
5. Press Insert+ spacebar key to navigate in browse mode. 
6. Navigate to the cards using arrow key. 
7. Observe the screen reader announcement.

### Actual behavior:
In Browse mode, when the focus is on card screen reader announces 'link' on every control and text.

### Expected behavior:
In Browse mode, the screen reader should not announce as 'link' on every control and text.

### NVDA logs, crash dumps and other attachments:
<img width="957" alt="In Browse mode, the screen reader announces 'link' on every control and text under 'Browse App' section" src="https://github.com/nvaccess/nvda/assets/93735775/cdce3acd-fc3a-4610-bb29-f9cfa0a73768">

https://github.com/nvaccess/nvda/assets/93735775/9519733c-0ad6-43be-b2bf-8376a405bf6a

#### Does the issue still occur after restarting your computer?
Yes

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
No

**Comment by Adriani90** [comment]
We need a minimal test case or codepen for this. The link you provided requires to have a Microsoft account to test this, and not everyone has a Microsoft account in this community.

**Comment by seanbudd** [comment]
We are unable to process this issue as it stands.

To be able to reproduce this issue, we need a code sample. Provide a minimal HTML sample to reproduce this using [codepen](https://codepen.io/).

Kind Regards,
NV Access Software Developers

**Comment by msftedad** [comment]
Hi @seanbudd, Sorry for the inconvenience. I have updated the repro steps with publicly accessible URL . Could you please verify.

**Comment by seanbudd** [comment]
This is intended behaviour. Each element in that grouping is a link, pressing enter opens the link. It is important that screen reader users are aware that a link is focused.
